From e610d427374226b79da5258d979936459f30c761 Mon Sep 17 00:00:00 2001
From: fluzz <fluzz@gmx.com>
Date: Wed, 13 May 2020 13:28:18 +0200
Subject: [PATCH] gcc10 fails to build, due to 'multiple definition errors'.

The '-fno-common' option is now a default for gcc, and so the linker
can no more resolve multiple tentative definitions.

This patch removes those multiple definitions.
This fixes issue965, reported by the Debian package maintainer.
---
 src/global.h                | 2 +-
 src/lvledit/lvledit_tools.h | 4 ++--
 src/mapgen/mapgen.h         | 6 +++---
 src/struct.h                | 2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/global.h b/src/global.h
index 8159f73d9..39dee7c37 100644
--- a/src/global.h
+++ b/src/global.h
@@ -36,7 +36,6 @@ extern char *freedroid_version;
 #define INTERN_FOR _main_c
 #include "extint_macros.h"
 
-EXTERN float FPSover1;
 EXTERN char *AllSkillTexts[];
 EXTERN char font_switchto_red[];
 EXTERN char font_switchto_blue[];
@@ -95,6 +94,7 @@ EXTERN FPSmanager SDL_FPSmanager;
 #define INTERN_FOR _misc_c
 #include "extint_macros.h"
 
+EXTERN float FPSover1;
 EXTERN float timeout_from_item_drop;
 EXTERN int use_open_gl;
 #ifdef DEBUG_QUAD_BORDER
diff --git a/src/lvledit/lvledit_tools.h b/src/lvledit/lvledit_tools.h
index 0c451b552..25920aa2e 100644
--- a/src/lvledit/lvledit_tools.h
+++ b/src/lvledit/lvledit_tools.h
@@ -34,9 +34,9 @@ struct leveleditor_tool {
 	int (*display) ();
 };
 
-void leveleditor_init_tools(void);
+EXTERN void leveleditor_init_tools(void);
 
-struct leveleditor_tool tool_place, tool_move, tool_select;
+EXTERN struct leveleditor_tool tool_place, tool_move, tool_select;
 
 #define EVENT_LEFT_PRESS(e) (((e) && e->type == SDL_MOUSEBUTTONDOWN) && (e->button.button == 1))
 #define EVENT_RIGHT_PRESS(e) (((e) && e->type == SDL_MOUSEBUTTONDOWN) && (e->button.button == 3))
diff --git a/src/mapgen/mapgen.h b/src/mapgen/mapgen.h
index d7ac10010..edccdb6bc 100644
--- a/src/mapgen/mapgen.h
+++ b/src/mapgen/mapgen.h
@@ -64,9 +64,9 @@ extern struct roominfo *rooms;
 extern int total_rooms;
 
 // Interface to the game
-void (*dungeonmap_convert) (int, int, unsigned char *);
-void (*dungeonmap_place_enemies) (struct roominfo *);
-void (*dungeonmap_gift) (struct roominfo *);
+//void (*dungeonmap_convert) (int, int, unsigned char *);
+//void (*dungeonmap_place_enemies) (struct roominfo *);
+//void (*dungeonmap_gift) (struct roominfo *);
 
 int generate_dungeon_gram(int, int);
 
diff --git a/src/struct.h b/src/struct.h
index b44618b34..d98ee28cf 100644
--- a/src/struct.h
+++ b/src/struct.h
@@ -1063,7 +1063,7 @@ struct tux_rendering {
 	struct tux_part_instances default_instances;    // Default part instances
 	tux_part_render_motionclass *render_order;      // The render_sets of each motion class
 	int gun_muzzle_height;							// Vertical offset to apply to bullets
-} tux_rendering;
+};
 
 /**
  * Contains a set of Tux's parts images for a motion class.
-- 
GitLab

